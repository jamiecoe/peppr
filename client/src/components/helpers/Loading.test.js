import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

describe('Loading', () => {
  const mockRetry = jest.fn();
  const props = {
    error: null,
    isLoading: true,
    pastDelay: false,
    retry: mockRetry,
    timedOut: false,
  };

  let loading = shallow(<Loading {...props} />);

  it('should render properly', () => {
    expect(loading).toMatchSnapshot();
  });

  describe('when there is an error on props', () => {
    beforeEach(() => {
      props.error = 'test error';
      loading = shallow(<Loading {...props} />);
    });

    afterEach(() => {
      props.error = null;
    });

    it('should render properly', () => {
      expect(loading).toMatchSnapshot();
    });

    it('should call retry() method on props, when button is clicked', () => {
      loading.find('button').simulate('click');
      expect(mockRetry).toHaveBeenCalled();
    });
  });

  describe('when Loading has timed out', () => {
    beforeEach(() => {
      props.timedOut = true;
      loading = shallow(<Loading {...props} />);
    });

    afterEach(() => {
      props.timedOut = false;
    });

    it('should render properly', () => {
      expect(loading).toMatchSnapshot();
    });

    it('should call retry() method on props, when button is clicked', () => {
      loading.find('button').simulate('click');
      expect(mockRetry).toHaveBeenCalled();
    });
  });

  describe('when delay is longer than your set delay', () => {
    beforeEach(() => {
      props.pastDelay = true;
      loading = shallow(<Loading {...props} />);
    });

    afterEach(() => {
      props.pastDelay = false;
    });

    it('should render properly', () => {
      expect(loading).toMatchSnapshot();
    });
  });
});
