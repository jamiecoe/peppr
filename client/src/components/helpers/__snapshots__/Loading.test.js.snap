// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    error={null}
    isLoading={true}
    pastDelay={false}
    retry={[Function]}
    timedOut={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Loading when Loading has timed out should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    error={null}
    isLoading={true}
    pastDelay={false}
    retry={[Function]}
    timedOut={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Taking a long time... ",
        <button
          onClick={[Function]}
>
          Retry
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Taking a long time... ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retry",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Retry",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Taking a long time... ",
          <button
            onClick={[Function]}
>
            Retry
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Taking a long time... ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retry",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Retry",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Loading when delay is longer than your set delay should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    error={null}
    isLoading={true}
    pastDelay={true}
    retry={[Function]}
    timedOut={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Loading when there is an error on props should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    error="test error"
    isLoading={true}
    pastDelay={false}
    retry={[Function]}
    timedOut={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Error! ",
        <button
          onClick={[Function]}
>
          Retry
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Error! ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retry",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Retry",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Error! ",
          <button
            onClick={[Function]}
>
            Retry
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Error! ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retry",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Retry",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
